sbor/src/decode.rs

```diff
@@ -86,9 +88,21 @@ impl<'de, X: CustomTypeId> Decoder<'de, X> {
     }
 
     pub fn read_size(&mut self) -> Result<usize, DecodeError> {
-        let mut bytes = [0u8; 4];
-        bytes.copy_from_slice(self.read_slice(4)?);
-        Ok(u32::from_le_bytes(bytes) as usize)
+        // LEB128 and 4 bytes max
+        let mut size = 0usize;
```

Comment:
We possibly don't care, but this would silently overflow on 16-bit machines...

Is it worth some const assert that usize is at least u32 size here?

Reply:
Come on, 16-bit machines.